libs/langgraph/src/pregel/index.ts

```diff
@@ -1318,7 +1318,10 @@ export class Pregel<
       if (updates.length === 1) {
         // eslint-disable-next-line prefer-const
         let { values, asNode } = updates[0];
-        if (asNode === undefined && nonNullVersion === undefined) {
+        if (asNode === undefined && Object.keys(this.nodes).length === 1) {
```

Comment:
ubernit but personal pref is one asNode === undefined` and then conditions (Object.keys(this.nodes).length === 1, else if nonNullVersion === undefined) else

Reply:
I tend to agree, but I think for this particular code it's probably better to keep it inlined - the nesting complexity is already really high and I find it hard to keep track of conditions that applied in some block 100+ lines up when I'm reading a bunch of conditional logic like this